_source/_posts/2022-05-02-angular-microfrontend-auth.md

```diff
@@ -0,0 +1,443 @@
+
+So, your e-commerce site using micro-frontends to decompose different functionality might look like this diagram, where the shopping cart and account features are in their separate routes within your Single Page Application (SPA):
+
+{% img blog/angular-microfrontend-auth/example-site-diagram.jpg alt:"Image of the example balloon e-commerce site showing a breakdown of the views. The entire site is wrapped in a host named 'Shell', and micro-frontends for the 'Cart' and 'Account' links" width:"800" %}{: .center-image }
+
+You might be saying, "Micro-frontends sound cool, but managing the different front-ends and orchestrating state across the micro-frontends also sounds complicated," and you're right. The concept of a micro-frontend has been around for a few years and rolling your own micro-frontend implementation, shared state, and tools to support it _was_ quite an undertaking. However, micro-frontends are now well supported with Webpack 5 and [Module Federation](https://webpack.js.org/concepts/module-federation/). Not all web apps require a micro-frontend architecture, but for those large, feature-rich web apps that were starting to get unwieldy, the first-class support of micro-frontends in our web tooling is definitely a plus.
+
+This post is part one of a series where we'll build an e-commerce site using Angular and micro-frontends. We'll use Webpack 5 with Module Federation to support wiring the micro-frontends together, demonstrate sharing authenticated state between the different front-ends, and deploy it all to a free cloud hosting provider. 
+
+In this first post, we'll explore a starter project and understand how the different apps connect, add authentication using Okta, and add the wiring for sharing authenticated state. In the end, you'll have an app that looks like this:
+
+{% img blog/angular-microfrontend-auth/final-app.gif alt:"Animated gif showing the final project with micro-frontends that handle the cupcake shopping basket, authentication, and profile information" width:"800" %}{: .center-image }
+
+**Prerequisites**
+
+- [Node](https://nodejs.org/) This project was developed using Node v16.14 with npm v8.5 
+- [Angular CLI](https://angular.io/cli)
+- [Okta CLI](https://cli.okta.com)
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+
+## Angular app with micro-frontend starter using Webpack and Module Federation
+
+There's a lot in this web app! We'll use a starter code to make sure we focus on the micro-frontend-specific code. If you're dismayed that you're using a starter and not starting from scratch, don't worry. I'll provide the Angular CLI commands to recreate the structure of this starter app on the repository's README.md so you have all the instructions.
```

Comment:
```suggestion
There's a lot in this web app! We'll use a starter code to make sure we focus on the code that's specific to the micro frontend. If you're dismayed that you're using a starter and not starting from scratch, don't worry. I'll provide the Angular CLI commands to recreate the structure of this starter app on the repository's README.md so you have all the instructions.
```
to avoid getting tripped up in hyphens in the 2nd sentence